mirage/tar_mirage.ml

```diff
@@ -349,17 +349,17 @@ module Make_KV_RW (CLOCK : Mirage_clock.PCLOCK) (BLOCK : Mirage_block.S) = struc
         let open Int64 in
         let sector_size = of_int t.info.Mirage_block.sector_size in
 
-        let start_bytes = sub t.end_of_archive 512L in
-        let header_start_bytes = sub start_bytes 512L in
-        let sentinel = 2 * 512 in
-        let end_bytes = add start_bytes (add space_needed (of_int sentinel)) in
+        let start_bytes = sub t.end_of_archive (of_int Tar.Header.length) in
+        let header_start_bytes = sub start_bytes (of_int Tar.Header.length) in
+        let sentinel = mul 2L (of_int Tar.Header.length) in
+        let end_bytes = add start_bytes (add space_needed sentinel) in
```

Comment:
after these computations:
- end_of_archive is behind the sentinel (2 * header_length, all zeroes)
- start_bytes is where the first piece of data will be written (end_of_archive - header_length)
- end_bytes is start_bytes (so the new tar header is already accounted for) + space_needed (the data, padded to header_length) + 2 * header_length (sentinel)